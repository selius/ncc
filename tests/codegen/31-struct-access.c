int main()
{
	struct {
		int a;
		float b;
	} s;

	struct {
		int a;
		struct {
			int a;
		} b;
		float c;
	} c;

	s.a = 5;
	s.b = 3.78;

	__print_int(s.a);
	__print_float(s.b);

	c.b.a = 99;

	__print_int(c.b.a);

	c.a = c.b.a;
	__print_int(c.a);
	
	return 0;
}
