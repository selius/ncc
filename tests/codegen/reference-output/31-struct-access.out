5
3.780000
99
99
